UKHomeOffice/nmsw-ui
# NMSW-538 Add submitted tag and submission date (and other status tags/hidden status)

# Ticket



----

## AC

Given I am viewing a draft declaration
When I am on the check your answers page
Then I am shown no status line
And the call to action is the submit panel and button

Given I am viewing a Submitted or PreSubmitted declaration
When I am on the check your answers page
Then I am shown a submitted tag
And the call to action is a cancel button

Given I am viewing a PreCancelled or Cancelled declaration
When I am on the check your answers page
Then I am shown a cancelled tag
And there is no call to action panel

Given I am viewing a Failed declaration
When I am on the check your answers page
Then I am shown a failed tag
And there is no call to action panel

----

## To test

- create a declaration
- go to the CYA page
- > you should NOT see a status line at the top of the voyage details
- > you SHOULD see the submit text and buttons
- submit the declaration
- go to the CYA page
- > you should see a status line with submitted tag and submission date at the top
- > you should see a cancel button at the bottom (it does not function yet)
- > you should NOT see a submit button
- > you should NOT see the submit supporting text

NOTE: the UI puts the declaration into a 'Draft' or 'PreSubmitted' status only (and PreCancelled when we build the cancel button function)
If we mock the other statuses these are the expected outcomes:

**Submitted**

> you should see a status line with submitted tag and submission date at the top
> you should see a cancel button at the bottom
> you should NOT see a submit button

**Cancelled**

> you should see a status line with cancelled tag and submission date at the top
> you should NOT see a cancel button
> you should NOT see a submit button
> you should NOT see the submit supporting text

**Failed**

> you should see a status line with Failed tag and submission date at the top
> you should NOT see a cancel button
> you should NOT see a submit button
> you should NOT see the submit supporting text

----

## Notes
